src/merge-exports.js

```diff
@@ -112,6 +115,30 @@ export async function mergeExports(options) {
         logger(`${result.wrappers.length} wrappers created`);
     }
 
+    if (options.preserveExtensions) {
+        const mappedExports = {};
+        const mappedTypes = {};
+
+        for (const key in result.pkg.exports) {
```

Comment:
i used `for..in` here because i was getting inconsistent ordering between test runs when using `Object.entries` and `Object.fromEntries` and i think this should preserve order

Reply:
even running the tests in serial i get this issue. not sure what the cause is